test/subscription_map.hpp

```diff
@@ -258,54 +302,72 @@ class subscription_map_base {
 
     // Find all topic filters and allow modification
     template<typename Output>
-    void modify_match(MQTT_NS::string_view topic, Output callback) {
+    void modify_match(MQTT_NS::string_view topic, Output&& callback) {
         find_match_impl(*this, topic, std::forward<Output>(callback));
     }
 
-    template<typename Output>
-    void handle_to_iterators(handle h, Output output) {
+    template<typename ThisType, typename Output>
+    static void handle_to_iterators(ThisType& self, handle const &h, Output&& output) {
         auto i = h;
-        while(true) {
-            if (i == root_key) {
-                return;
-            }
-
-            auto entry_iter = map.find(i);
-            if (entry_iter == map.end()) {
-                throw std::runtime_error("Invalid handle was specified");
+        while(i != self.root_key) {
+            auto entry_iter = self.map.find(i);
+            if (entry_iter == self.map.end()) {
+                throw_invalid_handle();
             }
 
             output(entry_iter);
             i = entry_iter->second.parent;
         }
     }
 
+    // Exceptions used
+    static void throw_invalid_topic_filter() { throw std::runtime_error("Subscription map invalid topic filter was specified"); }
+    static void throw_invalid_handle() { throw std::runtime_error("Subscription map invalid handle was specified"); }
+    static void throw_max_stored_topics() { throw std::overflow_error("Subscription map maximum number of stored topic filters reached"); }
+    static void throw_internal_error() { throw std::runtime_error("Subscription map internal error"); }
+
     // Get the iterators of a handle
-    std::vector<map_type_iterator> handle_to_iterators(handle h) {
+    std::vector<map_type_iterator> handle_to_iterators(handle const &h) {
```

Comment:
This function can be const.

Reply:
Make method const but still return modifyable iterator?

Reply:
If you want the returned iterators to be mutable then make two versions of the function. One const one non-const.